### PR TITLE
[ENTESB-4920] fix Java 8 test fails, use ECJ compatible with Java 8

### PR DESCRIPTION
**Pull request by psiroky**
* this is a temporary fix until SY upgrades to ip-bom 6.0.3.Final
   (it currently brings in ECJ 4.3.x which is not Java 8 compatible)

### CODE REVIEW

**Comment by psiroky** [comment]
@dvirgiln please try locally with Java 8. These changes should workaround the fails.

**Comment by dvirgiln** [comment]
This is great!

File: quickstarts/switchyard-bpm-service/pom.xml [hunk 1]
```diff
@@ -102,6 +102,12 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <!-- TODO: remove this override once SwitchYard upgrades to ip-bom 6.0.3.Final -->
+            <dependency>
+                <groupId>org.eclipse.jdt.core.compiler</groupId>
+                <artifactId>ecj</artifactId>
+                <version>4.4.2</version>
+            </dependency>
```
Comment (dvirgiln) [inline quickstarts/switchyard-bpm-service/pom.xml]: Normally we do not include any version in any pom file. All the dependency versions are defined on the main pom.xml dependency management section

File: quickstarts/switchyard-demo-library/pom.xml [hunk 2]
```diff
@@ -98,6 +98,12 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <!-- TODO: remove this override once SwitchYard upgrades to ip-bom 6.0.3.Final -->
+            <dependency>
+                <groupId>org.eclipse.jdt.core.compiler</groupId>
+                <artifactId>ecj</artifactId>
+                <version>4.4.2</version>
```
Comment (dvirgiln) [inline quickstarts/switchyard-demo-library/pom.xml]: Normally we do not include any version in any pom file. All the dependency versions are defined on the main pom.xml dependency management section

File: quickstarts/switchyard-bpm-service/pom.xml (hunk 1 above)
Comment (psiroky) [inline quickstarts/switchyard-bpm-service/pom.xml]: True, but this is a quickstart which does not have any parent...

Comment (dvirgiln) [inline quickstarts/switchyard-bpm-service/pom.xml]: :) yes! 

Well we can then merge this!

**Pull request merged**